internal/providers/amazon/objectstore.go

```diff
@@ -180,19 +184,43 @@ func (s *objectStore) DeleteBucket(bucketName string) error {
 	bucket := &ObjectStoreBucketModel{}
 	searchCriteria := s.searchCriteria(bucketName)
 
-	logger.Info("looking for bucket")
+	logger.Infof("looking up bucket %s", bucketName)
 
 	if err := s.db.Where(searchCriteria).Find(bucket).Error; err != nil {
 		if err == gorm.ErrRecordNotFound {
 			return bucketNotFoundError{}
 		}
 	}
 
-	logger.Info("deleting bucket")
+	if err := s.deleteFromProvider(bucket); err != nil {
+		if !s.force {
+			// if delete is not forced return here
+			return err
+		}
+	}
+
+	db := s.db.Delete(bucket)
+	if db.Error != nil {
+		return s.deleteFailed(bucket, db.Error)
+	}
+
+	return nil
+}
+
+func (s *objectStore) deleteFromProvider(bucket *ObjectStoreBucketModel) error {
+	logger := s.getLogger().WithField("bucket", bucket.Name)
+	logger.Infof("deleting bucket %s on provider", bucket.Name)
```

Comment:
The bucket name is already in the logger's context so we may not need to log it here again.